samples/dotnet/KernelHttpServer/Plugins/GitHubPlugin.cs

```diff
@@ -102,55 +72,48 @@ public GitHubSkill(IKernel kernel, ILogger<GitHubSkill>? logger = null)
     /// <summary>
     /// Summarize the code downloaded from the specified URI.
     /// </summary>
-    /// <param name="source">URI to download the repository content to be summarized</param>
-    /// <param name="context">Semantic kernel context</param>
-    /// <returns>Task</returns>
-    [SKFunction("Downloads a repository and summarizes the content")]
-    [SKFunctionName("SummarizeRepository")]
-    [SKFunctionInput(Description = "URL of the GitHub repository to summarize")]
-    [SKFunctionContextParameter(Name = RepositoryBranchParamName,
-        Description = "Name of the repository repositoryBranch which will be downloaded and summarized")]
-    [SKFunctionContextParameter(Name = SearchPatternParamName, Description = "The search string to match against the names of files in the repository")]
-    public async Task SummarizeRepositoryAsync(string source, SKContext context)
+    /// <param name="input">URI to download the repository content to be summarized</param>
+    /// <param name="repositoryBranch">Name of the repository repositoryBranch which will be downloaded and summarized</param>
+    /// <param name="searchPattern">The search string to match against the names of files in the repository</param>
+    /// <param name="patToken">Personal access token for private repositories</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Name of the memory collection used to store the code summaries.</returns>
+    [Description("Downloads a repository and summarizes the content")]
+    public async Task<string?> SummarizeRepositoryAsync(
```

Comment:
I am seeing this error:

> Microsoft.SemanticKernel.KernelException: Function not available: Function not available GitHubPlugin.SummarizeRepository

With the `SKFunctionName` attribute removed, how does this function get associated with the name `SummarizeRepository`?

Reply:
There's logic in there that's supposed to find the function with and without the "Aync" suffix, but it might not be working as I expected. I'll add the SKName attributes here and the other functions. Thanks for testing, and good find!

Reply:
@gitri-ms Updated -- can you try again please?

I'll also get my environment set up to run these too. Thanks again!